src/ai/BaseAI.js

```diff
@@ -172,6 +177,8 @@ export const BaseAI = class {
         this.stats.requests.attempts++;
         start = (new Date()).getTime();
 
+        this.signal.addEventListener('abort', hardStopListener);
```

Comment:
if (this.signal) ...

or this.signal?.removeEventListener(...) // preferred if this syntax works, please check it